plugin/documents.py

```diff
@@ -285,6 +287,12 @@ def on_diagnostics_updated_async(self, is_view_visible: bool) -> None:
         is_active_view = window and window.active_view() == self.view
         if is_active_view and self.view.change_count() == self._change_count_on_last_save:
             self._toggle_diagnostics_panel_if_needed_async()
+        self._diagnostics_view.clear_annotations()
+        if userprefs().show_diagnostics_annotations:
+            all_diagnostics = []  # type: List[Tuple[Diagnostic, sublime.Region]]
+            for _, diagnostics in self._diagnostics_async(allow_stale=True):
+                all_diagnostics.extend(diagnostics)
+            self._diagnostics_view.update_diagnostic_annotations_async(all_diagnostics)
```

Comment:
I guess it might be more efficient and potentially less flickery if we would do that from SessionView, per-server. Like we do highlight regions currently.